src/v/datalake/translation/partition_translator.cc

```diff
@@ -313,10 +313,28 @@ partition_translator::do_translate_once(retry_chain_node& parent_rcn) {
 ss::future<partition_translator::checkpoint_result>
 partition_translator::checkpoint_translated_data(
   retry_chain_node& rcn,
+  kafka::offset reader_begin_offset,
   coordinator::translated_offset_range translated_range) {
     if (translated_range.files.empty()) {
         co_return checkpoint_result::yes;
     }
+    if (reader_begin_offset != translated_range.start_offset) {
```

Comment:
I'm wondering what happens if we had no records, even if we read to the end? E.g. if all the data was aborted. Should we still send an offset range?

Reply:
In that case, translation attempt is marked failure with [no data](https://github.com/redpanda-data/redpanda/blob/9a9a4ae834ddf9c9b08788373a79999cb3e86af4/src/v/datalake/record_multiplexer.cc#L190) and we retry again until atleast one data batch is processed. So IIUC, only start offset needs to be adjusted.